### PR TITLE
[linux-port] Remaining cleanups.

### PR DESCRIPTION
**Pull request by ehsannas**
A few things that were overdue for a clean up. It also makes it easier to post to the upstream repo.

### CODE REVIEW

File: lib/DxcSupport/WinFunctions.cpp [hunk 1]
```diff
@@ -51,7 +51,19 @@ HRESULT IntToUInt(int in, UINT *out) {
     hr = S_OK;
   } else {
     *out = 0xffffffff;
-    hr = (HRESULT)1L;
+    hr = ERROR_ARITHMETIC_OVERFLOW;
```
Comment (pow2clk) [inline lib/DxcSupport/WinFunctions.cpp]: Should  `UIntAdd` above not return this error as well?

File: lib/DxcSupport/WinFunctions.cpp [hunk 2]
```diff
@@ -51,7 +51,19 @@ HRESULT IntToUInt(int in, UINT *out) {
...
+}
+HRESULT SizeTToInt(size_t in, int *out) {
+  HRESULT hr;
+  if(in <= INT_MAX) {
+    *out = (int)in;
+    hr = S_OK;
+  }
+  else {
+    *out = 0xffffffff;
+    hr = ERROR_ARITHMETIC_OVERFLOW;
```
Comment (pow2clk) [inline lib/DxcSupport/WinFunctions.cpp]: Also UInt32Mult below.

File: lib/HLSL/DxilRootSignature.cpp [hunk 3]
```diff
@@ -726,10 +723,7 @@ void RootSignatureVerifier::VerifyRootSignature(
       // Verify root descriptor flags.
       unsigned Flags = (unsigned)pSlot->Descriptor.Flags;
       if (Flags & ~(unsigned)DxilRootDescriptorFlags::ValidFlags) {
-        // TODO(ehsann): Looks like llvm::DiagnosticPrinter does not play well with std::hex.
-        /*
```
Comment (pow2clk) [inline lib/HLSL/DxilRootSignature.cpp]: Oh yeah. I missed this.

File: tools/clang/tools/dxcompiler/dxcompilerobj.cpp [hunk 4]
```diff
@@ -748,6 +749,7 @@ class DxcCompiler : public IDxcCompiler2,
       ::llvm::sys::fs::MSFileSystem *msfPtr;
       IFT(CreateMSFileSystemForDisk(&msfPtr));
       std::unique_ptr<::llvm::sys::fs::MSFileSystem> msf(msfPtr);
+
```
Comment (pow2clk) [inline tools/clang/tools/dxcompiler/dxcompilerobj.cpp]: This file is all whitespace changes. I'm pretty indifferent to whitespace changes, but unless I'm mistaken the blanks being added were just removed. If there's reason for this, I'm already on board whatever it is. I just want to verify it's not included by mistake.

File: lib/DxcSupport/WinFunctions.cpp (hunk 1 above)
Comment (ehsannas) [reply lib/DxcSupport/WinFunctions.cpp]: ah yes. missed that.

File: lib/DxcSupport/WinFunctions.cpp (hunk 2 above)
Comment (ehsannas) [reply lib/DxcSupport/WinFunctions.cpp]: yeah. missed that.

File: tools/clang/tools/dxcompiler/dxcompilerobj.cpp (hunk 4 above)
Comment (ehsannas) [reply tools/clang/tools/dxcompiler/dxcompilerobj.cpp]: yeah, it's intentional. I'm bringing down the diff between the MS and google TOT.
We are down to only a few hundred lines of difference. Tempting to post all as one CL :laughing:  (But i won't of course). I'll be sending a few CLs to MS repo today to further bring down the difference. Hoping to reach zero difference soon (depending on how fast our PRs are reviewed).

**Pull request merged**